src/evtx_chunk.rs

```diff
@@ -307,39 +314,41 @@ impl<'a> Debug for EvtxChunk<'a> {
 }
 
 impl EvtxChunkHeader {
-    pub fn from_reader(input: &mut Cursor<&[u8]>) -> Result<EvtxChunkHeader, failure::Error> {
+    pub fn from_reader(input: &mut Cursor<&[u8]>) -> Result<EvtxChunkHeader> {
         let mut magic = [0_u8; 8];
-        input.take(8).read_exact(&mut magic)?;
+        input.take(8).read_exact(&mut magic).context(err::IO)?;
 
-        if &magic != b"ElfChnk\x00" {
-            return Err(format_err!(
-                "Wrong chunk header magic {:?}, magic, expected ElfChnk\x00",
-                &magic
-            ));
-        }
+        ensure!(
+            &magic == b"ElfChnk\x00",
+            err::InvalidEvtxChunkMagic { magic }
+        );
 
-        let first_event_record_number = input.read_u64::<LittleEndian>()?;
-        let last_event_record_number = input.read_u64::<LittleEndian>()?;
-        let first_event_record_id = input.read_u64::<LittleEndian>()?;
-        let last_event_record_id = input.read_u64::<LittleEndian>()?;
+        let first_event_record_number = try_read!(input, u64);
```

Comment:
Liked it better before, because now it's less clear that you are reading in LittleEndian, and it looks different from lines 344-351 for no good reason really

Reply:
Everything is little endian in all windows stuff, but I wanted to reduce the context capturing boilerplate (offset extraction).